digdag-core/src/main/java/io/digdag/core/workflow/WorkflowExecutor.java

```diff
@@ -342,6 +342,18 @@ private void noticeStatusPropagate()
         }
     }
 
+    public void noticeRunWhileConditionChange()
```

Comment:
Yes. It is necessary to shorten shutdown duration that returns 503 Service Temporarily Unavailable errors to REST clients.
`runWhile` method sleeps MAX_INTERVAL (5 seconds) at most if there're no actively updating tasks. It uses `propagatorCondition.await(time)` to sleep. noticeRunWhileConditionChange breaks the sleep by sending notification to `propagatorCondition`.